libtransmission/torrent.h

```diff
@@ -82,6 +84,8 @@ struct tr_incomplete_metadata;
 struct tr_torrent final : public tr_completion::torrent_view
 {
 public:
+    using labels_t = small::set<tr_quark, 0>;
```

Comment:
This is a really good point.

At first I got caught up on the terminology of "ordered set" and "unordered set" since that is similar to the C++ terminology of `std::set`, `std::unordered_set`, `small:set`, `small::unordered_set`.

But I think you're talking about "the order they fields are added in" a la #3271 and you're absolutely right about that point and I'd forgotten about it. The semantics were "sorted by the user's order preference", rather than alphabetically or by tr_quark's integer value. So https://github.com/transmission/transmission/issues/5598 doesn't really make sense. :disappointed: 

If this PR were merged as-is it would reintroduce the same bug that #3271 fixed. @Coeur nice catch.